### PR TITLE
Fix type names in `pytools.tag`

### PR DESCRIPTION
**Pull request by inducer**
Upon reviewing https://github.com/inducer/arraycontext/pull/154, I found myself very much not loving the type names I was looking at. @alexfikl, could you take a look if this seems sane? (I'll do a pytools release once some version of this is in.)

### CODE REVIEW

File: pytools/tag.py [hunk 1]
```diff
@@ -162,9 +162,9 @@ class UniqueTag(Tag):
 # }}}
 
 
-TagsType = FrozenSet[Tag]
-TagOrIterableType = Union[Iterable[Tag], Tag, None]
-T_co = TypeVar("T_co", bound="Taggable")
+TagSet = FrozenSet[Tag]
```
Comment (alexfikl) [inline pytools/tag.py:165]: Is this required? Writing `FrozenSet[Tag]` seems clearer and not that long.

**Review by alexfikl — APPROVED** [review]
Definitely better!

File: pytools/tag.py [hunk 2]
```diff
@@ -162,9 +162,9 @@ class UniqueTag(Tag):
...
+ToTagSetConvertible = Union[Iterable[Tag], Tag, None]
+Taggable_co = TypeVar("Taggable_co", bound="Taggable")
```
Comment (alexfikl) [inline pytools/tag.py:167]: Should this have a `TypeVar(..., covariant=True)`?

File: pytools/tag.py (hunk 1 above)
Comment (inducer) [reply pytools/tag.py]: Good point. Dropped.

File: pytools/tag.py (hunk 2 above)
Comment (inducer) [reply pytools/tag.py]: Upon further thought, what these IMO really should be is the `Self` type from Py 3.11.

Comment (inducer) [reply pytools/tag.py]: (I've dropped the type variable, with deprecation.)

Comment (alexfikl) [reply pytools/tag.py]: I didn't look before how it's used, but yeah, that makes sense!

**Comment by inducer** [comment]
Thanks for taking a look!

cc @nchristensen

**Pull request merged**